File: Easy/attivform/meta_attivformappelloview/meta_attivformappelloview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_attivformappelloview
{
    public class Meta_attivformappelloview : Meta_easydata 
	{
        public Meta_attivformappelloview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "attivformappelloview") {
				Name = "Attività formative";
			EditTypes.Add("appello");
			ListingTypes.Add("appello");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"aa","idattivform","idcorsostudio","iddidprog","iddidproganno","iddidprogcurr","iddidprogori","iddidprogporzanno"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "appello": {
						DescribeAColumn(T, "didprog_title", "Didattica programmata", nPos++);
						DescribeAColumn(T, "insegn_denominazione", "Denominazione Insegnamento", nPos++);
						DescribeAColumn(T, "insegn_codice", "Codice Insegnamento", nPos++);
						DescribeAColumn(T, "insegninteg_denominazione", "Denominazione Integrando", nPos++);
						DescribeAColumn(T, "insegninteg_codice", "Codice Integrando", nPos++);
						DescribeAColumn(T, "attivform_tipovalutaz", "Profitto o Idoneità", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "appello": {
						return "attivform_sortcode";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/attivform/meta_attivformappelloview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_attivformappelloview() {
        MetaData.apply(this, ["attivformappelloview"]);
        this.name = 'meta_attivformappelloview';
    }

    meta_attivformappelloview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_attivformappelloview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'appello':
						this.describeAColumn(table, 'title', 'Attività formativa', null, 1000, -1);
						this.describeAColumn(table, 'aa', 'AA', null, 2000, 9);
						this.describeAColumn(table, 'didproganno_anno', 'Anno di corso', null, 3100, null);
						this.describeAColumn(table, 'didprog_title', 'Didattica programmata', null, 5100, 1024);
						this.describeAColumn(table, 'didprog_aa', 'Anno accademico Didattica programmata', null, 5200, 9);
						this.describeAColumn(table, 'sede_title', 'Sede Didattica programmata', null, 5320, 1024);
						this.describeAColumn(table, 'attivform_tipovalutaz', 'Profitto o Idoneità', null, 18000, null);
//$objCalcFieldConfig_appello$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["aa", "idsede", "iddidprog", "idattivform", "iddidprogori", "idcorsostudio", "iddidproganno", "iddidprogcurr", "iddidprogporzanno"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "appello": {
						return "";
					}
					case "appello": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('attivformappelloview', new meta_attivformappelloview('attivformappelloview'));

	}());
